### ISSUE TITLE
Weak delegates not correctly identified with Swift 4.1

### ISSUE THREAD

**Issue by dbarden** [realm/SwiftLint#2019]
### New Issue Checklist

- [X] Updated SwiftLint to the latest version
- [X] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

With Swift 4.1 that came out with Xcode 9.3 beta, the weak delegates don't seem to be identified correctly.

```bash
Linting Swift files in current working directory
Linting 'atest.swift' (1/1)
/Users/barden/projects/DTI/swiftlint-test/atest.swift:10:8: warning: Weak Delegate Violation: Delegates should be weak to avoid reference cycles. (weak_delegate)
Done linting! Found 1 violation, 0 serious in 1 file.
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)?
`0.24.2`
* Installation method used (Homebrew, CocoaPods, building from source, etc)?
`homebrew`

* Which Xcode version are you using (check `xcode-select -p`)?

/Applications/Xcode-9.3.app/Contents/Developer (9.3 beta from 24th Jan)

* Do you have a sample that shows the issue? 

```swift
protocol TestDelegate: class {

}

protocol TestProtocol {
  weak var delegate: TestDelegate? { get }
}

class Test {
  weak var delegate: TestDelegate
}
```

[Here](https://gist.github.com/dbarden/3727c106404b2015db765b86a8b5263a) there is the sourkitten structure output as well.

**Comment by dbarden** [comment]
Thanks :)

**Comment by MillerApps** [comment]
Apparently it is still an issue. Happens on the Xcode 9.3 release. Installed through Coocapods.

**Comment by marcelofabri** [comment]
@MillerApps please fill a new issue following the issue template

**Comment by diejmon** [comment]
@marcelofabri When can we expect a release with this bug fix?

**Comment by MillerApps** [comment]
@marcelofabri looks like the pod for SwiftLint was out of date.

**Comment by BilalReffas** [comment]
Yes I have the same issue on Xcode 9.3

**Comment by MarcAMartin** [comment]
Seeing this issue as well

**Comment by marcelofabri** [comment]
Please open a new issue following the template and making sure you're on the latest version of SwiftLint.

**Comment by diejmon** [comment]
Make sure you have the latest swiftlint: `swiftlint version`. Should `0.25.1` by now.
